keps/sig-node/4381-dra-structured-parameters/README.md

```diff
@@ -1188,239 +937,309 @@ drivers are expected to be written for Kubernetes.
+const ResourceSliceMaxAttributesPerDevice = 32
+
+// ResourceSliceMaxDevices and ResourceSliceMaxAttributesPerDevice where chosen
+// so that with the maximum attribute length of 96 characters the total size of
+// the ResourceSlice object is around 420KB.
+
+// DeviceAttribute is a combination of an attribute name and its value.
+// Exactly one value must be set.
+type DeviceAttribute struct {
+    // Name is a unique identifier for this attribute, which will be
+    // referenced when selecting devices.
+    //
+    // Attributes are defined either by the owner of the specific driver
+    // (usually the vendor) or by some 3rd party (e.g. the Kubernetes
+    // project). Because attributes are sometimes compared across devices,
+    // a given name is expected to mean the same thing and have the same
+    // type on all devices.
+    //
+    // Attribute names must be either a C-style identifier
+    // (e.g. "the_name") or a DNS subdomain followed by a slash ("/")
+    // followed by a C-style identifier
+    // (e.g. "example.com/the_name"). Attributes whose name does not
+    // include the domain prefix are assumed to be part of the driver's
+    // domain. Attributes defined by 3rd parties must include the domain
+    // prefix.
+    //
+    // The maximum length for the DNS subdomain is 63 characters (same as
+    // for driver names) and the maximum length of the C-style identifier
+    // is 32.
     Name string
 
-    NamedResourcesAttributeValue
-}
+    // The Go field names below have a Value suffix to avoid a conflict between the
+    // field "String" and the corresponding method. That method is required.
+    // The Kubernetes API is defined without that suffix to keep it more natural.
 
-// NamedResourcesAttributeValue must have one and only one field set.
-type NamedResourcesAttributeValue struct {
     // QuantityValue is a quantity.
     QuantityValue *resource.Quantity
     // BoolValue is a true/false value.
     BoolValue *bool
-    // IntValue is a 64-bit integer.
-    IntValue *int64
-    // IntSliceValue is an array of 64-bit integers.
-    IntSliceValue *NamedResourcesIntSlice
-    // StringValue is a string.
+    // StringValue is a string. Must not be longer than 64 characters.
     StringValue *string
-    // StringSliceValue is an array of strings.
-    StringSliceValue *NamedResourcesStringSlice
     // VersionValue is a semantic version according to semver.org spec 2.0.0.
+    // Must not be longer than 64 characters.
     VersionValue *string
 }
 
-// NamedResourcesIntSlice contains a slice of 64-bit integers.
-type NamedResourcesIntSlice struct {
-    // Ints is the slice of 64-bit integers.
-    Ints []int64
-}
-
-// NamedResourcesStringSlice contains a slice of strings.
-type NamedResourcesStringSlice struct {
-    // Strings is the slice of strings.
-    Strings []string
-}
+const DeviceAttributeMaxIDLength = 32
+const DeviceAttributeMaxValueLength = 64
```

Comment:
What are these exactly? We've never talked about an Attribute ID...

Reply:
An attribute name consists of a domain prefix and an identifier part. That's what I meant with "ID" here.

Reply:
OK, its still confusing to me without context though since we never use this name anywhere else. I'm also not quite sure what the Attribute Value is -- does this only apply to the `string` based attributes?

Reply:
Added comments.